test/upgrade/prober/wathola/receiver/types.go

```diff
@@ -18,7 +18,9 @@ package receiver
 
 // Report represents state as JSON
 type Report struct {
-	State  string   `json:"state"`
-	Events int      `json:"events"`
-	Thrown []string `json:"thrown"`
+	State         string   `json:"state"`
+	Events        int      `json:"events"`
+	Duplicated    []string `json:"duplicated"`
+	Missing       []string `json:"missing"`
+	DefaultThrown []string `json:"defaultThrow"`
```

Comment:
I propose of having:

```suggestion
	Thrown `json:"thrown"`
```
and:

```golang
struct Thrown struct {
  Duplicated   []string `json:"duplicated"`
  Other        []string `json:"defaultThrow"`
}
```
As we have only special case for duplicates. Or as I originally propose to have dedicated errors like:

```golang
type Thrown struct {
  Duplicated []string
  Missing    []string
  Excess     []string
}
```

So, solution which have a sense is either:

 * to have as many types as different processes (duplicates has separate process than others)
 * to have as many types as they are, and then process them according to their type (this is slightly more open to future changes)

Reply:
Fixed as suggested, use your original `unexpected`.